tests/sync_client_test.cc

```diff
@@ -0,0 +1,36 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "sync_client.h"
+
+class SyncClientTest : public ::testing::Test {
+  public:
+    std::string GetRequestStr(const std::string& host,
+        const std::string& uri) {
+      return "GET " + uri + " HTTP/1.1\r\n" \
+        "User-Agent: curl/7.35.0\r\n" \
+        "Accept: */*\r\n" \
+        "Host: " + host + "\r\n" \
+        "Connection: close\r\n" \
+        "\r\n";
+    }
+};
+
+TEST_F(SyncClientTest, IntegratedSanity)
+{
+  SyncClient client;
+  std::string response;
+
+  // site 1
+  ASSERT_TRUE(client.Connect("ucla.edu", "80"));
+  ASSERT_TRUE(client.Write(GetRequestStr("ucla.edu", "/")));
```

Comment:
I know it will usually be fine, but you may not want your unit tests to depend on whether ucla.edu is up at the time.